modules/passkey/contracts/4337/SafeWebAuthnSharedSigner.sol

```diff
@@ -49,6 +44,23 @@ contract SafeWebAuthnSharedSigner is SignatureValidator {
      */
     uint256 public immutable SIGNER_SLOT;
 
+    /**
+     * @notice Emitted when the shared signer is configured for an account.
+     * @dev Note that the configured account is not included in the event data. Since configuration
+     * is done as a `DELEGATECALL`, the contract emitting the event is the configured account. This
+     * is also why the event name is prefixed with `SafeWebAuthnSharedSigner`, in order to avoid
+     * event `topic0` collisions with other contracts (seeing as "configured" is a common term).
+     * @param x The x-coordinate of the public key.
+     * @param y The y-coordinate of the public key.
+     * @param verifiers The P-256 verifiers to use.
+     */
+    event SafeWebAuthnSharedSignerConfigured(uint256 x, uint256 y, P256.Verifiers verifiers);
```

Comment:
My first question when I saw this was: do we need them to be indexed or not? I recall we discussed and agreed that it shouldn't be, but perhaps it's worth a comment?

Reply:
Documented in the PR description.

I didn't include it in a comment to match what we did for `Created` event in the other contract. If you think this _should_ be comments, then I will do it in a separate PR and add comments for both events.

Reply:
I'm sorry if we've already talked about this, but let's say that given x and y coordinates, I want to find a safe where this passkey is configured as a signer. Would that be possible without indexed events?

IIRC, we concluded that this was possible without indexing, but I forgot how :see_no_evil:

Reply:
We said it was possible based on address for the proxy signer (you assume some standard `verifiers` configuration that you use for your Dapp and search for the `Created` event by `signer` field), but I don't think it is possible for this event at the moment.

One issue you would have as well, is that the event isn't emitted by a single contract but by the Safe contract itself (because of the `DELEGATECALL`), so you would have to search by topic0 and topic1 for all contracts (don't think that is an issue - just worth pointing out).

If we think "index Safes with this WebAuthn public key" is a usecase, then we should add a "public key hash" indexed field (that is computed as `keccak256(abi.encode(x, y))` or something similar) to make this easier without requiring two indexed fields.

Reply:
> We said it was possible based on address for the proxy signer (you assume some standard verifiers configuration that you use for your Dapp and search for the Created event by signer field)

But can anyone emit the `Created` event? I can directly create a signer without creating a Safe, can't I? So it's rather a non-specific one.

> so you would have to search by topic0 and topic1 for all contracts (don't think that is an issue - just worth pointing out).

Ah, this is a significant constraint, IMO. My use case was pretty much compatibility with autonomous Safe interfaces that don't require a backend, but it doesn't seem that adding this event will make it particularly friendly for this use case.

Reply:
I'm a little lost. This is about the shared signer contract, right? So, no signer will actually be created, so you cannot use `AddOwner`. Do we have an alternative option then?

Regarding the validity of the use case, I do think it's very useful for trustless interfaces. We can ask the broader team for an opinion

Reply:
Sorry, let me try to clarify - the `Created` event is only emitted by the factory contract. The first group steps above are for "I want to find a safe where this passkey is configured as a signer" use case for the **factory** contract, and not the shared signer:

> I think we would then fetch AddOwner events by the signer that we derive the address for.

The second group of steps are for the same use case, but this time with the shared signer:

> For the shared signer, you could do with a index on `heccak256(abi.encode(x, y))`:

Sorry for the confusion - I just imagined that the "I want to find a safe where this passkey is configured as a signer" needs to be achievable from **both** versions of the signers (both factory+proxy and shared), so I highlighted steps from each of them.

Reply:
OK - I'm going to merge this PR as is, and address this comment in a separate PR (adding a user story to show this intended use case).

Reply:
https://github.com/safe-global/safe-modules/pull/463